iaas/amazonec2/amazonec2.go

```diff
@@ -0,0 +1,152 @@
+package amazonec2
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"context"
+
+	"github.com/docker/machine/drivers/amazonec2"
+	"github.com/docker/machine/libmachine"
+	"github.com/docker/machine/libmachine/drivers/rpc"
+	"github.com/docker/machine/libmachine/host"
+	"github.com/gofn/gofn/iaas"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Provider definition, represents a concrete implementation of an iaas
+type Provider struct {
+	Client     libmachine.API
+	Host       *host.Host
+	Name       string
+	ClientPath string
+	Region     string
+	Size       string
+	ImageSlug  string
+	KeyID      int
+	Ctx        context.Context
+}
+
+type driverConfig struct {
+	DriverName string `json:"DriverName"`
+	Driver     struct {
+		InstanceID  string `json:"InstanceID"`
+		MachineName string `json:"MachineName"`
+		IPAddress   string `json:"IPAddress"`
+		AMI         string `json:"AMI"`
+		SSHKeyID    int    `json:"SSHKeyID"`
+	} `json:"Driver"`
+}
+
+func getConfig(machineDir, hostName string) (config *driverConfig, err error) {
+	configPath := fmt.Sprintf("%s/%s/config.json", machineDir, hostName)
+	raw, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return
+	}
+	err = json.Unmarshal(raw, &config)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func setFlags(driver *amazonec2.Driver) (err error) {
+	flags := driver.GetCreateFlags()
+
+	driverOpts := rpcdriver.RPCFlags{
+		Values: make(map[string]interface{}),
+	}
+
+	// conversion of flags extracted from:
+	// https://github.com/docker/machine/blob/master/commands/create.go#L354-L361
+	for _, f := range flags {
+		driverOpts.Values[f.String()] = f.Default()
+
+		// Hardcoded logic for boolean... :(
+		if f.Default() == nil {
+			driverOpts.Values[f.String()] = false
+		}
+	}
+	// TODO: receive this configs to remove hard coded values
```

Comment:
why have this todo ?

Reply:
It's not the ideal scenario to have this configurations hard coded, instead, we should receive them as parameters, or (my preferred option) get the default values just like the docker-machine binary uses when creating a machine via cli, but honestly, I wasn't able to replicate this behavior.